### ISSUE TITLE
Migrations try to make columns nullable that already are

### ISSUE THREAD

**Issue by LiteracyFanatic** [efcore/EFCore.FSharp#104]
**Describe the bug**
Consider the following simple model.

```fsharp
[<CLIMutable>]
type Test = {
    Id: int
    Date: DateTime option
}
```
```fsharp
modelBuilder.Entity<Test>(fun entity ->

    entity.ToTable("Test")
        |> ignore

    entity.Property(fun e -> e.Id)
        .IsRequired()
        |> ignore

    entity.Property(fun e -> e.Date)
        .IsRequired(false)
        .HasColumnType("datetime")
        |> ignore
) |> ignore
```
The `Up` method in the initial migration correctly creates the table with the `Date` column being nullable.

```fsharp
override this.Up(migrationBuilder:MigrationBuilder) =
    migrationBuilder.CreateTable(
        name = "Test"
        ,columns = (fun table -> 
        {|
            Id =
                table.Column<int>(
                    nullable = false
                    ,``type`` = "int"
                ).Annotation("SqlServer:Identity", "1, 1")
            Date =
                table.Column<DateTime>(
                    nullable = true // This is correct
                    ,``type`` = "datetime"
                )
        |})
        ,constraints =
            (fun table -> 
                table.PrimaryKey("PK_Test", (fun x -> (x.Id) :> obj)) |> ignore
            ) 
    ) |> ignore
```

However, `BuildTargetModel` in the migration and `BuildModel` in the snapshot don't recognize the `Date` column as being nullable.

```fsharp
override this.BuildTargetModel(modelBuilder: ModelBuilder) =
    modelBuilder

        .UseIdentityColumns().HasAnnotation("Relational:Collation", "SQL_Latin1_General_CP1_CI_AS")
        .HasAnnotation("Relational:MaxIdentifierLength", 128)
        .HasAnnotation("ProductVersion", "5.0.7")
        |> ignore

    modelBuilder.Entity("ef_test.TestDomain+Test", (fun b ->

        b.Property<int>("Id")
            .ValueGeneratedOnAdd()
            .HasColumnType("int")
            .UseIdentityColumn() |> ignore

        // This is wrong
        b.Property<DateTime>("Date")
            .HasColumnType("datetime") |> ignore

        b.HasKey("Id") |> ignore

        b.ToTable("Test") |> ignore

    )) |> ignore
```

As a result, when diffing with the snapshot to create a new migration, EF thinks it needs to alter the `Date` column in the database to make it nullable (even though it already is).

```fsharp
override this.Up(migrationBuilder:MigrationBuilder) =
    migrationBuilder.AlterColumn<DateTime>(
        name = "Date"
        ,table = "Test"
        ,``type`` = "datetime"
        ,nullable = true
        ,oldClrType = typedefof<DateTime>
        ,oldType = "datetime"
        ,oldNullable = false // In reality it already is nullable, but the snapshot is wrong so EF thinks it needs to update the column
        ) |> ignore
```

This is mostly harmless, but multiplied across many tables it fills the migration files with a lot of noise that makes them difficult to review.

**Expected behavior**
Creating a new migration without changing the model should result in an empty `Up` method.

Adding a call to `IsRequired(false)` in the migration and snapshot resolves the issue. Changing `<DateTime>` to `<DateTime option>` is needed to stop `IsRequired(false)` from throwing a runtime error (and is also the correct type of the property).

```fsharp
b.Property<DateTime option>("Date")
    .IsRequired(false)
    .HasColumnType("datetime") |> ignore
```

This can also be avoided by use `Nullable<DateTime>` instead of `DateTime option` but I'd like to avoid that if possible.

**Comment by simon-reynolds** [comment]
Resolved by #107 
Will release 5.0.3-beta005 in the next few minutes

**Comment by LiteracyFanatic** [comment]
Awesome! This should make migrations much easier to review. Thanks so much.
